### ISSUE TITLE
E402 ignores dunder variables between imports

### ISSUE THREAD

**Issue by chaush-server** [astral-sh/ruff#16815]
### Summary

```python
from asyncio import gather

__all__ = ("RuleAccessor",)

from code.base.rule.accessor import RuleBaseAccessor
```

![Image](https://github.com/user-attachments/assets/dd0c4e2d-f66e-43ff-95ef-2afb4381a8e8)

**Comment by MichaReiser** [comment]
`__dunder__` names were excluded in https://github.com/astral-sh/ruff/pull/294

because of

> Module level "dunders" (i.e. names with two leading and two trailing
underscores) such as all, author, version, etc. should be
placed after the module docstring but before any import statements
except from future imports.

There's an argument that we should flag `__all__` in between imports but that would be a different rule.
